From e3cd68bc0870c58545efe24084cceb4d43683c3d Mon Sep 17 00:00:00 2001
From: Hugo Osvaldo Barrera <hugo@whynothugo.nl>
Date: Thu, 14 May 2026 16:29:46 +0200
Subject: [PATCH 2/2] Dynamically link system libcurl-impersonate

This patch is Alpine-specific, and not suitable for upstream.
---
 scripts/build.py | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/scripts/build.py b/scripts/build.py
index 13cfa9a..49847a8 100644
--- a/scripts/build.py
+++ b/scripts/build.py
@@ -64,7 +64,7 @@ def detect_arch():
     obj_name = (
         "libcurl-impersonate.dll" if system == "Windows" else "libcurl-impersonate.a"
     )
-    link_type = "dynamic" if system == "Windows" else "static"
+    link_type = "dynamic"
     sysname = SYSNAME_MAP.get(system)
 
     if system == "Android":
@@ -93,6 +93,7 @@ def detect_arch():
     if libdir:
         libdir = os.path.expanduser(libdir)
 
+    libdir = "/usr/lib"
     return {
         "system": system,
         "machine": machine,
@@ -142,6 +143,10 @@ print(f"Using {libdir} to store libcurl-impersonate")
 
 def download_libcurl():
     expected = libdir / obj_name
+    if is_dynamic:
+        print("libcurl-impersonate: dynamically linking system library")
+        return
+
     if expected.exists():
         print(f"libcurl-impersonate: {expected} already downloaded.")
         return
@@ -262,7 +267,7 @@ ffibuilder.set_source(
     include_dirs=[
         str(root_dir / "include"),
         str(root_dir / "ffi"),
-        str(libdir / "include"),
+        "/usr/include",
     ],
     sources=[
         str(root_dir / "ffi/shim.c"),
-- 
2.55.0

